Add the go version in Github pipeline

**Issue/Feature Description:**
Add the go version in the Github pipeline, by default it is using 1.15 but the vdo project needs 1.16

**What type of Issue/Feature is this?**
/kind bug fix

**Why this fix/ feature is needed(give scenarios or use cases):**
Pipelines are failing as the go version(1.15) is not compatible with the few packages 

**How to reproduce, in case of a bug:**
From [here](https://github.com/vmware-tanzu/vsphere-kubernetes-drivers-operator/runs/4161411284?check_suite_focus=true)
```
go: downloading github.com/google/go-cmp v0.5.6
# github.com/golangci/golangci-lint/internal/robustio
Error: /home/runner/go/pkg/mod/github.com/golangci/golangci-lint@v1.43.0/internal/robustio/robustio_other.go:18:9: undefined: os.ReadFile
note: module requires Go 1.16
# github.com/golangci/golangci-lint/pkg/fsutils
Error: /home/runner/go/pkg/mod/github.com/golangci/golangci-lint@v1.43.0/pkg/fsutils/filecache.go:27:20: undefined: os.ReadFile
note: module requires Go 1.16
# github.com/go-critic/go-critic/checkers
Error: /home/runner/go/pkg/mod/github.com/go-critic/go-critic@v0.6.1/checkers/ruleguard_checker.go:149:17: undefined: os.ReadFile
# github.com/breml/bidichk/pkg/bidichk
Error: /home/runner/go/pkg/mod/github.com/breml/bidichk@v0.1.1/pkg/bidichk/bidichk.go:46:15: undefined: os.ReadFile
note: module requires Go 1.16
make[1]: *** [Makefile:320: golangci_lint] Error 2
make: *** [Makefile:292: lint] Error 2
make[1]: Leaving directory '/home/runner/work/vsphere-kubernetes-drivers-operator/vsphere-kubernetes-drivers-operator'
Error: Process completed with exit code 2.
```

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

--- a/.github/workflows/BuildAndDeploy.yml
+++ b/.github/workflows/BuildAndDeploy.yml
@@ -22,7 +22,6 @@ jobs:
         uses: actions/setup-go@v2.1.4
         with:
           go-version: '1.16'
-        run: go version
 
       - name: Build and Deploy
         run: make deploy

--- a/.github/workflows/BuildAndDeploy.yml
+++ b/.github/workflows/BuildAndDeploy.yml
@@ -20,6 +20,9 @@ jobs:
 
       - name: Setup Go environment
         uses: actions/setup-go@v2.1.4
+        with:
+          go-version: '1.16'
+        run: go version
 
       - name: Build and Deploy
         run: make deploy

